packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -774,21 +760,52 @@ export class ContainerRuntime
 
 		const mc = loggerToMonitoringContext(logger);
 
+		// Some options require a minimum version of the FF runtime to operate, so the default configs will be generated
+		// based on the compatibility mode.
+		// For example, if compatibility mode is set to "1", the default configs will ensure compatibility with FF runtime
+		// 1.x  or later. If the compatibility mode is set to "2", the default values will be generated to ensure compatibility
+		// with FF runtime 2.x or later.
+		// Our policy is to support N/N-1 compatibility by default, where N is the most recent public major release of the runtime.
+		// Therefore, if the customer does not provide a compatibility mode, we will default to use N-1.
+		const defaultCompatibilityMode = "1"; // TODO: Automatically set to be N-1.
+		const compatibilityMode = runtimeOptions.compatibilityMode ?? defaultCompatibilityMode;
+		const defaultConfigs = getConfigsForCompatMode(compatibilityMode);
+
+		// Here we assign each container runtime option to the value provided in `runtimeOptions`.
+		// We also will assign the default values generated above if the user does not provide a value.
```

Comment:
Can you do something with object spread to start with the default config and cover it with runtime options, rather than all the defaults spelled out on individual lines?  If you make the type of default config have everything required* then you should be able to do 

```
const {
  foo,
  bar,
  baz
} = { ...defaultConfigs, runtimeOptions }
```

\* If there's some explicitly left off default configs, then you'll have to put defaults for those, but you can put in on the RHS of the `=`.  This is very subjective, but IMO it would be more readable.  If it even pans out like I'm picturing.

Reply:
Yeah I will definetely clean up the syntax. I think for now I will wait until we figure out what we are doing about the "grandfathered" configs (i.e. group batching, flush mode), since those currently behave differently and it will be a bit messy.